draft-sullivan-tls-opaque.md

```diff
@@ -89,59 +107,69 @@ when, and only when, they appear in all capitals, as shown here.
-
-TLS 1.3 provides a KCI-secure key agreement algorithm suitable for use with OPAQUE. This document describes three instantiations of OPAQUE in TLS 1.3: one based on digital signatures, one on Diffie-Hellman key agreement, and one based on HMQV key exchange. Of the three instantiations, the only one that has known IPR considerations is HMQV.
-
-OPAQUE consists of two distinct phases: password registration and authentication. We will describe the mechanisms for password registration in this document but it is assumed to have been done outside of TLS. During password registration, the client and server establish a shared set of parameters for future authentication and two private-public key pairs are generated, one for the client and one for the server. The server keeps its private key and stores an encapsulated copy of the client’s key pair along with its own public key in an “envelope” that is encrypted with the result of the OPRF operation. Note that it is possible for the server to use the same key for multiple clients. It may be necessary to permit multiple simultaneous server keys in the even of a key rollover. The client does not store any state nor any PKI information.
-
-We call the first instantiation OPAQUE-Sign. In OPAQUE-Sign, the key pairs generated at password registration time are digital signature keys. These signature keys are used in place of certificate keys for both server and client authentication in a TLS handshake. Client authentication is technically optional, but in practice is almost universally required. OPAQUE-Sign cannot be used alongside certificate-based handshake authentication. This instantiation can also be leveraged to do part of a post-handshake authentication using Exported Authenticators {{!I-D.ietf-tls-exported-authenticator}} given an established TLS connection protected with certificate-based authentication.
+OPAQUE {{opaque-paper}} is a Strong Asymmetric Password-Authenticated Key Exchange (Strong aPAKE) built
+on an oblivious pseudo-random function (OPRF) and authenticated key exchange protocol that is secure
+against key-compromise impersonation (KCI) attacks. Unlike previous PAKE methods such as SRP {{?RFC2945}}
+and SPAKE-2 {{?I-D.irtf-cfrg-spake2}}, which require a public salt value, a Strong aPAKE leverages the
+OPRF private key as salt, making it resistant to pre-computation attacks on the password database stored on the server.
+
+TLS 1.3 provides a KCI-secure key agreement algorithm suitable for use with OPAQUE. This document describes two
+instantiations of OPAQUE in TLS 1.3: one based on digital signatures, called OPAQUE-Sign, and one on Diffie-Hellman
+key agreement, called OPAQUE-3DH.
+
+OPAQUE consists of two distinct phases: password registration and authentication. We will describe the
```

Comment:
This is actually a point that I was wanting to chat with you: while the protocol focuses on password, it could be used for anything kind of information, right? @chris-wood

Reply:
Hmm, what would you use if not passwords?